### PR TITLE
Add message comparison helpers to auv_interfaces package

### PR DESCRIPTION
**Pull request by hidmic**
# Description

Precisely what the title says. These helpers aid stagnation checks in publishers.

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [x] Unit tests are passing
- [x] Linter tests are passing

# How To Test

Run unit tests for `auv_interfaces` package:

```sh
catkin_make run_tests_auv_interfaces
```

### CODE REVIEW

File: auv_interfaces/package.xml [hunk 1]
```diff
@@ -14,4 +14,5 @@
   <depend>geometry_msgs</depend>
   <depend>std_msgs</depend>
   <exec_depend>message_runtime</exec_depend>
+  <test_depend>roslint</test_depend>
```
Comment (patapufate) [inline auv_interfaces/package.xml:17]: Should we add <test_depend>rosunit</test_depend> ?

**Review by patapufate — APPROVED** [review]
Despite the comment, LGTM.

File: auv_interfaces/package.xml (hunk 1 above)
Comment (hidmic) [reply auv_interfaces/package.xml:17]: Yes! Good catch. See https://github.com/ChrisScianna/ROS-Underwater-RnD/pull/93/commits/42ad2d2ba38bd98e1f49517263d0b29b1b0f9af7.

**Comment by hidmic** [comment]
Thanks for the review! Merging.

**Pull request merged**